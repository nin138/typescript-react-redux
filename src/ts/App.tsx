import * as History from 'history';
import * as React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Router } from 'react-router-dom';
import { HomeContainer } from './pages/home/Home';
import { store } from './store';
const history = History.createBrowserHistory();

export const App = () => {
  return (
    <Provider store={store}>
      <Router history={history}>
        <BrowserRouter>
          <Route path="/" component={HomeContainer} />
        </BrowserRouter>
      </Router>
    </Provider>
  );
};
