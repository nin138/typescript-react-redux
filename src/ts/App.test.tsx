import * as React from 'react';
import renderer, { act } from 'react-test-renderer';
import { App } from './App';

it('App component', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON()!;
  expect(tree).toMatchSnapshot();

  act(() => tree.children!.find(it => it.type === 'input')!.props.onChange({ target: { value: 'test text' } }));
  // re-rendering
  tree = component.toJSON()!;
  expect(tree).toMatchSnapshot();
});
