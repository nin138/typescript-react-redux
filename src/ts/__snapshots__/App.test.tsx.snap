// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<section
  className="app"
>
  <h1>
    Home
  </h1>
  <img
    alt="logo"
    height={16}
    src="./images/logo.svg"
    width={16}
  />
  hello, world!
  <br />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</section>
`;

exports[`App component 2`] = `
<section
  className="app"
>
  <h1>
    Home
  </h1>
  <img
    alt="logo"
    height={16}
    src="./images/logo.svg"
    width={16}
  />
  test text
  <br />
  <input
    onChange={[Function]}
    type="text"
    value="test text"
  />
</section>
`;
